mypy is confused by property aliases

Thank you very much for this very useful python package. Unfortunately mypy is very confused by property aliases such as 

```python
    @property
    def references(self) -> "IterableList[Reference]":
        """A list of :class:`~git.refs.reference.Reference` objects representing tags,
        heads and remote references.

        :return:
            ``git.IterableList(Reference, ...)``
        """
        return Reference.list_items(self)

    # Alias for references.
    refs = references
```

that can be found in [`git/repo/base.py`](https://github.com/gitpython-developers/GitPython/blob/4c21e514cd9b5acdc34891fc4b52e9b599810b3e/git/repo/base.py#L405C1-L416C22) or the analogous alias `branches` for `heads`. See https://github.com/python/mypy/issues/6700

You can reproduce the issue with the following `test.py`:
```python
from git.repo import Repo

repo = Repo()
for branch in repo.branches:
    print(branch)
```
about which `mypy` says:
```
test.py:4: error: "Callable[[], IterableList[Head]]" has no attribute "__iter__" (not iterable)  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

I am not aware of any fix that would not uglify the code here, and that mypy issue doesn’t seem likely to see improvement in the near future. This is sad, but doing all the work to add type annotations and still having users facing incomprehensible error messages in perfectly legitimate code is also sad. 

Could you please tell me whether you would welcome a PR “fixing” this?

--- a/git/remote.py
+++ b/git/remote.py
@@ -828,8 +828,15 @@ class Remote(LazyMixin, IterableObj):
             name._clear_cache()
         return name
 
-    # `rm` is an alias.
-    rm = remove
+    @classmethod
+    def rm(cls, repo: "Repo", name: str) -> str:
+        """Alias of remove.
+        Remove the remote with the given name.
+
+        :return:
+            The passed remote name to remove
+        """
+        return cls.remove(repo, name)
 
     def rename(self, new_name: str) -> "Remote":
         """Rename self to the given `new_name`.

--- a/git/repo/base.py
+++ b/git/repo/base.py
@@ -403,6 +403,17 @@ class Repo:
         return Head.list_items(self)
 
     @property
+    def branches(self) -> "IterableList[Head]":
+        """Alias for heads.
+        A list of :class:`~git.refs.head.Head` objects representing the branch heads
+        in this repo.
+
+        :return:
+            ``git.IterableList(Head, ...)``
+        """
+        return self.heads
+
+    @property
     def references(self) -> "IterableList[Reference]":
         """A list of :class:`~git.refs.reference.Reference` objects representing tags,
         heads and remote references.
@@ -412,11 +423,16 @@ class Repo:
         """
         return Reference.list_items(self)
 
-    # Alias for references.
-    refs = references
+    @property
+    def refs(self) -> "IterableList[Reference]":
+        """Alias for references.
+        A list of :class:`~git.refs.reference.Reference` objects representing tags,
+        heads and remote references.
 
-    # Alias for heads.
-    branches = heads
+        :return:
+            ``git.IterableList(Reference, ...)``
+        """
+        return self.references
 
     @property
     def index(self) -> "IndexFile":

